### PR TITLE
feat: use Entities as properties of other Entities

### PR DESCRIPTION
**Pull request by monitz87**
For example, if an Entity related to a Security Token had a ticker as a property before, it will
now have the SecurityToken object as a property instead

BREAKING CHANGE:
- replace `ticker: string` with `token: SecurityToken` in: `Checkpoint`, `CheckpointSchedule`,
  `CorporateAction`, `DividendDistribution`, `Sto`, `CustomPermissionGroup`,
  `KnownPermissionGroup`, `DefaultTrustedClaimIssuer`
- change the return type of `CustomPermissionGroup.toJson`. `id: BigNumber` is now `id: string`

### CODE REVIEW

File: src/api/entities/Checkpoint.ts [hunk 1]
```diff
@@ -240,10 +248,10 @@ export class Checkpoint extends Entity<UniqueIdentifiers, HumanReadable> {
    * Return the Checkpoint's ticker and identifier
    */
   public toJson(): HumanReadable {
-    const { ticker, id } = this;
+    const { token, id } = this;
 
     return toHumanReadable({
-      ticker,
+      ticker: token,
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/api/entities/Checkpoint.ts:254]: Indeed the token is serialized to a ticker

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**